### DISCUSSION TITLE
Notify on UMD app using options

### DISCUSSION THREAD

**Discussion by viniciusccarvalho** [quasarframework/quasar#14960] [General - Components / Directives / etc] [answered]
Hi folks, I'm struggling to get the Notify plugin working on my current setup. I'm using the options api:

```javascript
const { useQuasar } = Quasar
const { ref } = Vue
const $q = useQuasar()
const app = Vue.createApp({
  data() {
    return {
      model: ref(null)
    }
  },
  methods: {
    showNotif() {
      $q.notify({
          message: 'Jim pinged you.',
          color: 'purple'
        })
    }
  }
})

app.use(Quasar, { config: {} })
app.mount('#q-app')

        
```

But this is not working, not clear what am I missing here. All the examples on the site are using the composition api, and it was not clear to me how to integrate with my options version.

Thanks

**Comment by suleimanas** [comment]
if you are using UMD:
`Quasar.Notify.create('Hi and welcome!')`
https://quasar.dev/start/umd#quasar-global-object

**Comment by yusufkandemir** [comment, marked as answer]
I suspect you are mixing Composition API and Options API since you are trying to use `useQuasar` and you are using `ref` inside `data`. First of all, I would suggest learning the difference and using one or the other.

When using Options API, you can use `this.$q`:
```js
const app = Vue.createApp({
  data() {
    return {
      model: null
    }
  },
  methods: {
    showNotif() {
      this.$q.notify({
        message: 'Jim pinged you.',
        color: 'purple'
      })
    }
  }
})
```

When using Composition API:
```js
const { useQuasar } = Quasar
const { ref } = Vue

const app = Vue.createApp({
  setup() {
    const model = ref(null)

    const $q = useQuasar()
    function showNotif() {
      $q.notify({
        message: 'Jim pinged you.',
        color: 'purple'
      })
    }

    return {
      model,
      showNotif
    }
  }
})
```

In everywhere(_inside/outside components_), Options/Composition API:
```js
const { Notify } = Quasar

Notify.create({
  message: 'Jim pinged you.',
  color: 'purple'
})
```

**Reply by viniciusccarvalho** [reply]
Thanks
